service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -150,139 +176,136 @@ public void reset(final PolicyEvaluationRequestCacheKey key) {
+            map.put(subjectKey, timestamp);
         });
         multiSet(map);
     }
 
-    private void logSetSubjectTimestampDebugMessage(final String key, final String timestamp, final String subjectId) {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug(String.format("Setting timestamp for subject '%s'; key: '%s', value: '%s' ", subjectId, key,
-                    timestamp));
-        }
-    }
-
     private boolean isRequestCached(final List<String> values) {
-        if (null != values.get(values.size() - 1)) {
-            return true;
-        }
-        return false;
+        return null != values.get(values.size() - 1);
     }
 
     private boolean isCachedRequestInvalid(final List<String> values, final DateTime policyEvalTimestamp) {
         DateTime policyEvalTimestampUTC = policyEvalTimestamp.withZone(DateTimeZone.UTC);
 
-        if (connectorService.isResourceAttributeConnectorConfigured()
-                || connectorService.isSubjectAttributeConnectorConfigured()) {
-            return haveConnectorCacheIntervalsLapsed(connectorService, policyEvalTimestampUTC);
+        if (this.connectorService.isResourceAttributeConnectorConfigured()
+                || this.connectorService.isSubjectAttributeConnectorConfigured()) {
+            return haveConnectorCacheIntervalsLapsed(this.connectorService, policyEvalTimestampUTC);
         } else {
             return havePrivilegeServiceAttributesChanged(values, policyEvalTimestampUTC);
         }
     }
 
-    boolean havePrivilegeServiceAttributesChanged(final List<String> values,
-            final DateTime policyEvalTimestampUTC) {
-        for (int i = 0; i < values.size() - 1; i++) {
-            if (null == values.get(i)) {
-                continue;
+    boolean havePrivilegeServiceAttributesChanged(final List<String> values, final DateTime policyEvalTimestampUTC) {
+        for (String value : values) {
+            if (null == value) {
+                return true;
             }
             DateTime invalidationTimestampUTC;
             try {
-                invalidationTimestampUTC = (OBJECT_MAPPER.readValue(values.get(i), DateTime.class))
-                        .withZone(DateTimeZone.UTC);
+                invalidationTimestampUTC = (OBJECT_MAPPER.readValue(value, DateTime.class)).withZone(DateTimeZone.UTC);
             } catch (IOException e) {
                 throw new IllegalStateException("Failed to read timestamp from JSON.", e);
             }
-
             if (invalidationTimestampUTC.isAfter(policyEvalTimestampUTC)) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug(
+                            "Privilege service attributes have timestamp '%s' which is after "
+                                    + "policy evaluation timestamp '%s'",
+                            invalidationTimestampUTC, policyEvalTimestampUTC);
+                }
```

Comment:
Since you are using lazy string construction we could rely on debug to check the debug level. No need for extra isDebugEnabled call.  Will message formatting work with %s or should we use {}?